src/compiler/checker.ts

```diff
@@ -10996,20 +10996,22 @@ namespace ts {
         // constituents, in which case the isPartial flag is set when the containing type is union type. We need
         // these partial properties when identifying discriminant properties, but otherwise they are filtered out
         // and do not appear to be present in the union type.
-        function getUnionOrIntersectionProperty(type: UnionOrIntersectionType, name: __String): Symbol | undefined {
-            const properties = type.propertyCache || (type.propertyCache = createSymbolTable());
+        function getUnionOrIntersectionProperty(type: UnionOrIntersectionType, name: __String, skipObjectFunctionPropertyAugment?: boolean): Symbol | undefined {
+            const properties = skipObjectFunctionPropertyAugment ?
+                type.propertyCacheWithoutObjectFunctionPropertyAugment ||= createSymbolTable() :
+                type.propertyCache ||= createSymbolTable();
             let property = properties.get(name);

```

Comment:
This lookup should be able to be `type.propertyCacheWithoutObjectFunctionPropertyAugment?.get(name) || type.propertyCache?.get(name)`, since the later would otherwise fully duplicate the former (plus having the apparent members we intend to elide), since we'd prefer not to have two copies of a property symbol knocking about, no?